src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {Animated, View} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CalendarPicker from '../CalendarPicker';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import {propTypes, defaultProps} from './datepickerPropTypes';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import withNavigation from '../withNavigation';
+import compose from '../../libs/compose';
+import './styles.css';
+
+const datePickerPropTypes = {
+    ...propTypes,
+    ...windowDimensionsPropTypes,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedDate: null,
+            isPickerVisible: false,
+        };
+
+        this.setDate = this.setDate.bind(this);
+        this.togglePicker = this.togglePicker.bind(this);
+        this.removeClickListener = this.removeClickListener.bind(this);
+        this.handleOutsideClick = this.handleOutsideClick.bind(this);
+
+        this.opacity = new Animated.Value(0);
+        this.wrapperRef = React.createRef();
```

Comment:
this ref is required for web to close the calendar when clicked outside of the component

Reply:
Defining the ref in the constructor is redundant.

Reply:
in `handleOutsideClick` you already check if `wrapperRef` exist, how did you manage to get an error?

Reply:
I revisited the code I see you access current directly without checking `wrapperRef` itself, I have requested a change for that.

Reply:
After removing create ref, this.wrapperRef always is undefined when clicking outside

Reply:
Ah I see what you are doing wrong, well not wrong but different. In the `View` you have `ref={this.wrapperRef} ` this should be changed to a callback that set the ref to wrapperRef directly, i.e.:
```
ref={(ref) => {
    this.wrapperRef = ref;
}}
```

Also in `handleOutsideClick` you will have to access the ref directly, e.g. instead of `this.wrapperRef.current.abc` use `this.wrapperRef.abc`.

Reply:
Lets resolve this one, can we try what @s77rt proposed here?